pkgs/by-name/ma/maven/package.nix

```diff
@@ -2,13 +2,13 @@
   lib,
   callPackage,
   fetchurl,
-  jdk,
+  jdk_headless,
```

Comment:
It's common in Nixpkgs to do `maven.override { jdk = someJdk; }`, though. Could we move it to `all-packages.nix`?

Reply:
Can we use the @\doronbehar's trick? 

```
- jdk,
+ jdk_headless,
+ jdk ? jdk_headless,
```

Reply:
jdk is never null and checking if packages are null is an anti pattern and should not be done.

Reply:
> It's common in Nixpkgs to do `maven.override { jdk = someJdk; }`, though.

That is probably only required because we set JAVA_HOME in the wrapper. We shouldn't be required to rebuild the entire package just for that.

Reply:
`maven` is packaged from binary though, so the overhead isn't that big.

Reply:
It is still pretty unecessary if the fix is this easy [`ab554a9` (#326566)](https://github.com/NixOS/nixpkgs/pull/326566/commits/ab554a96a01e6aaaf90f1c36067d60ed2c978a2c)

Reply:
I agree.